yarn-project/aztec.js/src/wallet/account_wallet.ts

```diff
@@ -160,4 +198,31 @@ export class AccountWallet extends BaseWallet {
       returnTypes: [],
     };
   }
+
+  private getLookupValidityAbi(): FunctionAbi {
+    return {
+      name: 'lookup_validity',
+      isInitializer: false,
+      functionType: FunctionType.UNCONSTRAINED,
+      isInternal: false,
+      parameters: [
+        {
+          name: 'myself',
+          type: {
+            kind: 'struct',
+            path: 'authwit::aztec::protocol_types::address::aztec_address::AztecAddress',
+            fields: [{ name: 'inner', type: { kind: 'field' } }],
+          },
+          visibility: 'private' as ABIParameterVisibility,
+        },
+        {
+          name: 'block_number',
+          type: { kind: 'integer', sign: 'unsigned', width: 32 },
+          visibility: 'private' as ABIParameterVisibility,
+        },
+        { name: 'message_hash', type: { kind: 'field' }, visibility: 'private' as ABIParameterVisibility },
```

Comment:
Was just done as such to be similar to the others. Interesting to see that it seems to be somewhat ignored that stuff is missing in the unconstrained 🤣 My guess would almost be that  it is going straight for Fields if nothing specified then. But quite odd.

Reply:
Interestingly, if you are viewing it is not using the `encodeArguments` function but just blindly passing the arguments along so it does not use anything beyond the name of the function it seems 

```typescript
  public view(options: ViewMethodOptions = {}) {
    if (this.functionDao.functionType !== FunctionType.UNCONSTRAINED) {
      throw new Error('Can only call `view` on an unconstrained function.');
    }

    const { from } = options;
    return this.wallet.viewTx(this.functionDao.name, this.args, this.contractAddress, from);
  }
```

You can entirely delete the `parameters` and `returnTypes` from the abi and it will still succeed if unconstrained.